spaceship/lib/spaceship/tunes/tunes_client.rb

```diff
@@ -575,12 +575,25 @@ def candidate_builds(app_id, version_id)
     #####################################################
 
     # @param (testing_type) internal or external
-    def build_trains(app_id, testing_type, platform: nil)
+    def build_trains(app_id, testing_type, platform: nil, tries: 5)
       raise "app_id is required" unless app_id
       url = "ra/apps/#{app_id}/trains/?testingType=#{testing_type}"
       url += "&platform=#{platform}" unless platform.nil?
       r = request(:get, url)
-      parse_response(r, 'data')
+      return parse_response(r, 'data')
+    rescue Spaceship::Client::UnexpectedResponse => ex
+      # Build trains fail randomly very often
+      # we need to catch those errors and retry
+      # https://github.com/fastlane/fastlane/issues/6419
+      error_content = ex.to_s
+      if error_content == "{\"data\"=>nil, \"messages\"=>{\"warn\"=>nil, \"error\"=>[\"ITC.response.error.OPERATION_FAILED\"], \"info\"=>nil}, \"statusCode\"=>\"ERROR\"}"
+        unless (tries -= 1).zero?
+          logger.warn("Received temporary server error from iTunes Connect. Retrying the request...")
+          sleep 3 unless defined? SpecHelper
+          retry
+        end
+      end
+      return {} # better than nothing
```

Comment:
Oh yeah, great point, I wasn't thinking this through. Maybe we should even show a custom error message, that this might be a temporary server error 👍